OP-315 - Update startup script oh.sh for POH

mizzioisf opened the pull request:
Reworked the oh.sh startup script -  updated 23.10.2020

- More code polishing / reordering
- Added check to run as normal user
- Everything is now parametric and driven by shell variables that you can setup in the script (even more now :-)
- More comments, clearer console output
- Better error handling
- Added comments and blocks
- Better console / shell output for users
- Make the script PATH independent
- Reference MYSQL_DISTRO to remove hard-coded mysql version in etc/mysql/my.ori (now renamed in my.cnf.dist)
- Renamed references to the standard configuration file distributed (*.ori -> *.dist)
- NEW: If Java JRE or MySQL are not present, ask the user if he wants to download it
- NEW: command line options:
./oh.sh -h
Usage: oh.sh [-h -r]
   -h       shows this short help
   -r       reset Portable Open Hospital Installation

- NEW: possibility of resetting installation by archiving the database.sql create script into -> database.sql.imported
- created pull request for my.ori in order to make it variable independent

Note: in the default my.cnf config file I changed the default listening address fot MySQL to "localhost" instead of "0.0.0.0", since there is no reason to expose a POH database on the network.

mizzioisf committed:
Update oh.sh

mizzioisf committed:
Update oh.sh

mizzioisf committed:
Update my.ori

I define an external variable (MYSQL_DISTRO) which is defined in the proposed new oh.sh.
In this way we can keep this file mysql version indipendent

mizzioisf committed:
Update oh.sh

Added reference to MYSQL_DISTRO

mizzioisf committed:
Merge pull request #1 from mizzioisf/mizzioisf-patch-1

Update oh.sh

mizzioisf committed:
Rename dicom.properties.ori to dicom.properties.dist

I propose to rename all the template / standard configuration files bundle in the package, with the extension .dist, as in common usage.
If the change is accepted, I will modify the oh.sh script accordingly.

mizzioisf committed:
Rename log4j.properties.ori to log4j.properties.dist

I propose to rename all the template / standard configuration files bundle in the package, with the extension .dist, as in common usage.
If the change is accepted, I will modify the oh.sh script accordingly.

mizzioisf committed:
Update POH-README.md

- Added new supported dir
- Added quickinstall guide

mizzioisf committed:
Update POH-README.md

mizzioisf reviewed (commented):
Small update to the POH-README.md file

mizzioisf committed:
Update POH-README.md

mizzioisf reviewed (commented):
Line splitted.

mizzioisf committed:
Update oh.sh

mizzioisf reviewed (commented):
Moved MYSQL_PORT variable to the beginning of the file

mwithi committed:
Merge branch 'master' into master

mizzioisf committed:
Rename dicom.properties.ori to dicom.properties.dist

mizzioisf committed:
Rename my.ori to my.cnf.dist

mizzioisf committed:
Merge pull request #2 from mizzioisf/patch-5

Rename dicom.properties.ori to dicom.properties.dist

mizzioisf committed:
Merge pull request #3 from mizzioisf/patch-2

Patch 2

mizzioisf committed:
Merge pull request #4 from mizzioisf/patch-4

Rename log4j.properties.ori to log4j.properties.dist

mizzioisf committed:
Merge pull request #5 from mizzioisf/patch-3

Rename dicom.properties.ori to dicom.properties.dist

mizzioisf committed:
Update README.md

Open Hospital -> Portable Open Hospital

mizzioisf committed:
Update oh.sh

mizzioisf committed:
Update POH-README.md

mizzioisf committed:
Update POH-README.md

mizzioisf committed:
Update README.md

mizzioisf committed:
Update POH-README.md

mizzioisf committed:
Update oh.sh

mizzioisf committed:
Update my.cnf.dist

mizzioisf committed:
Merge pull request #6 from mizzioisf/mizzioisf-patch-OH.sh

Update oh.sh

mizzioisf committed:
Update POH-README.md

mizzioisf committed:
Update POH-README.md

mizzioisf committed:
Update oh.sh

mizzioisf committed:
Update oh.sh

mizzioisf committed:
Update POH-README.md

mizzioisf committed:
Update oh.sh

mizzioisf committed:
Update POH-README.md

Updated the Linux version part
Small cosmetic fixes

mizzioisf committed:
Update POH-README.md

mizzioisf committed:
Update oh.sh

mizzioisf committed:
Update oh.sh

Small cosmetic fixes to naming

mizzioisf committed:
Update oh.sh

Small cosmetic fixes

pviotti reviewed (approved):
Looks good to me. Left some comments on minor things and a doubt.

mizzioisf committed:
Update poh-bundle-linux-x64/oh.sh

Co-authored-by: Paolo <pviotti@users.noreply.github.com>

mizzioisf committed:
Update poh-bundle-linux-x64/oh.sh

Co-authored-by: Paolo <pviotti@users.noreply.github.com>

mizzioisf committed:
Update poh-bundle-linux-x64/oh.sh

Co-authored-by: Paolo <pviotti@users.noreply.github.com>

mizzioisf committed:
Update oh.sh

mizzioisf committed:
Update poh-bundle-linux-x64/oh.sh

Added error if script is run as root.

Co-authored-by: Paolo <pviotti@users.noreply.github.com>

mizzioisf committed:
Update oh.sh

Reordered block of code correcty.

mizzioisf committed:
Update oh.sh

Corrected check root user function

mizzioisf committed:
Update oh.sh

Renamed dicom.properties.ori to dicom.properties.dist (see related commit for file renaming)

mizzioisf committed:
Update oh.sh

Renamed log4j.properties.ori to log4j.properties.dist

mizzioisf reviewed (commented):
Everything should be included and fixed now...hopefully :-)
